source/solvers/postprocessing_cfd.cc

```diff
@@ -54,72 +54,116 @@ calculate_pressure_drop(const DoFHandler<dim> &       dof_handler,
+  double dynamic_pressure_outlet_boundary = 0.;
+  double dynamic_pressure_inlet_boundary  = 0.;
+  double dynamic_pressure_drop            = 0.;
+
+  double outlet_surface = 0.;
+  double inlet_surface  = 0.;
+
+  double temp_surface                      = 0.;
+  double temp_static_pressure_at_boundary  = 0.;
+  double temp_dynamic_pressure_at_boundary = 0.;
 
   for (const auto &cell : dof_handler.active_cell_iterators())
     {
       if (cell->is_locally_owned())
         {
-          fe_values.reinit(cell);
-          // Gather pressure (values)
-          fe_values[pressure].get_function_values(evaluation_point,
-                                                  present_pressure_values);
-
-          for (unsigned int q = 0; q < face_n_q_points; ++q)
+          for (unsigned int face_id = 0;
+               face_id < GeometryInfo<dim>::faces_per_cell;
+               face_id++)
             {
-              for (const auto &face : cell->face_iterators())
+              if (cell->face(face_id)->at_boundary())
                 {
-                  if (face->at_boundary() &&
-                      (face->boundary_id() == outlet_boundary_id))
+                  unsigned int boundary_id = cell->face(face_id)->boundary_id();
+                  if (boundary_id != inlet_boundary_id &&
+                      boundary_id != outlet_boundary_id)
+                    continue;
+                  fe_face_values.reinit(cell, face_id);
+                  fe_face_values[velocities].get_function_values(
+                    evaluation_point, velocity_values);
+                  fe_face_values[pressure].get_function_values(evaluation_point,
+                                                               pressure_values);
+                  temp_surface                      = 0.;
+                  temp_static_pressure_at_boundary  = 0.;
+                  temp_dynamic_pressure_at_boundary = 0.;
+                  for (unsigned int q = 0; q < face_n_q_points; q++)
                     {
-                      fe_face_values.reinit(cell, face);
-                      pressure_outlet_boundary +=
-                        present_pressure_values[q] * fe_face_values.JxW(q);
-
-                      outlet_surface += fe_face_values.JxW(q);
+                      temp_surface += fe_face_values.JxW(q);
+                      // Integration of velocity^2 and
+                      // pressure, since the pressure in Lethe has units of
+                      // Length^2/Time^2
+                      temp_static_pressure_at_boundary +=
+                        fe_face_values.JxW(q) * pressure_values[q];
+                      temp_dynamic_pressure_at_boundary +=
+                        fe_face_values.JxW(q) * velocity_values[q] *
+                        velocity_values[q];
                     }
-
-                  if (face->at_boundary() &&
-                      (face->boundary_id() == inlet_boundary_id))
+                  if (boundary_id == inlet_boundary_id)
```

Comment:
It is very weird that all your pressure drop from the static pressure as changes. I would try to find an explanation since from what I see here, it should be identical.

Reply:
I believe it is because the pressure values that were that were previously used came from the cell, and not the face (old line 76), even if the quadrature points iteration was done on the face. I reversed the concerned lines and tested on my machine and it seems to be the problem. 
From what I understand it would make more sense to take the values from the face like I did, even if it changes the values. @blaisb

Reply:
Yeah definitely it should be the values of the face and not of the cells